### PR TITLE
Median provider

### PR DESCRIPTION
**Pull request by connorwstein**

### CODE REVIEW

File: pkg/solana/contract_test.go [hunk 1]
```diff
@@ -148,15 +147,10 @@ func TestGetLatestTransmission(t *testing.T) {
 }
 
 func TestStatePolling(t *testing.T) {
-	i := atomic.NewInt32(0)
-	wait := 5 * time.Second
```
Comment (connorwstein) [inline pkg/solana/contract_test.go:154]: I don't think this is worth introducing a time dependence test just to test the polling for loop. Just made it explicit calls to get the values. Loop is tested in the integration tests anyways

File: pkg/solana/state_cache.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+)
+
+type StateCache struct {
+	utils.StartStopOnce
+	// on-chain program + 2x state accounts (state + transmissions)
+	ProgramID      solana.PublicKey
+	StateID        solana.PublicKey
+	StoreProgramID solana.PublicKey
+
+	stateLock sync.RWMutex
```
Comment (connorwstein) [inline pkg/solana/state_cache.go:32]: idiomatic to use non-pointers for the locks then just use pointers to the object holding the lock. Then you can make use of its default unlocked value

**Review by jmank88 — DISMISSED** [review]

File: pkg/solana/transmissions_cache.go [hunk 3]
```diff
@@ -32,17 +26,9 @@ type ContractTracker struct {
 	// private key for the transmission signing
 	Transmitter TransmissionSigner
```
Comment (aalu1418) [inline pkg/solana/transmissions_cache.go]: I think `Transmitter` and `TxManager` can be removed from the `TransmissionsCache` and `StateCache` - leftovers from the `ContractTracker` split

File: pkg/solana/transmissions_cache.go [hunk 4]
```diff
@@ -18,11 +16,7 @@ import (
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logger"
 )
 
-var (
-	configVersion uint8 = 1
-)
-
-type ContractTracker struct {
+type TransmissionsCache struct {
```
Comment (aalu1418) [inline pkg/solana/transmissions_cache.go:19]: Probably out of scope, but would be interesting if the cache could be generic between the Transmissions and State cache?

Comment (aalu1418) [reply pkg/solana/transmissions_cache.go:19]: further extension would be generic caching utility moved to `chainlink-relay`?

File: go.mod [hunk 5]
```diff
@@ -10,12 +10,10 @@ require (
 	github.com/gagliardetto/treeout v0.1.4
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.12.1
-	github.com/satori/go.uuid v1.2.0
-	github.com/smartcontractkit/chainlink v1.3.0
+	github.com/smartcontractkit/chainlink v1.4.1-rc1.0.20220512210926-5363811660bf
```
Comment (aalu1418) [inline go.mod]: Is this a stable commit on develop or a branch? (just concerned that a squash may leave this commit to be garbage collected by github sometime in the future)

Comment (connorwstein) [reply go.mod]: will need to do a follow up, since this is an interface change its a 3 PR process

File: pkg/solana/transmissions_cache.go (hunk 4 above)
Comment (connorwstein) [reply pkg/solana/transmissions_cache.go:19]: you could probably share the polling code, but thats pretty minimal and they may want to evolve independently anyways

**Review by jmank88 — DISMISSED** [review]

**Review by aalu1418 — DISMISSED** [review]

**Review by jmank88 — APPROVED** [review]

**Pull request merged**